pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java

```diff
@@ -450,18 +448,22 @@ public BrokerResponse handleRequest(JsonNode request, @Nullable RequesterIdentit
   private void updateQuerySource(BrokerRequest brokerRequest) {
     String tableName = brokerRequest.getQuerySource().getTableName();
     // Check if table is in the format of [database_name].[table_name]
-    String[] tableNameSplits = StringUtils.split(tableName, '.');
-    if (tableNameSplits.length != 2) {
-      return;
-    }
+    String[] tableNameSplits = StringUtils.split(tableName, ".", 2);
```

Comment:
Does the calcite parser not support getting columns names from [table].[column] format? If so, we should just use that, instead of post processing here? We have been adding small string manipulations per query incrementally, and I fear it will add up to have performance significance soon.

Reply:
Calcite doesn't support parse it as [table].[column], we will get the whole as identifier.
Then we need to handle that.

Reply:
If Calcite doesn't support, then should we support it? The logic seems brittle anyways.

Reply:
We should. The usage is for other system integrations like tableau